From a94f723df76ee431af944b6273868cf4c758eb6d Mon Sep 17 00:00:00 2001
From: aoand <aoand@mail.ru>
Date: Wed, 21 Aug 2019 19:26:45 +0300
Subject: [PATCH] fix syntax error in --proc-libs processing

libcheck(): Remove spurious braces from ${1} in an inline awk script
---
 checksec | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/checksec b/checksec
index 2329de8..e6bede3 100755
--- a/checksec
+++ b/checksec
@@ -632,7 +632,7 @@ proccheck() {
 # check mapped libraries
 libcheck() {
   ${debug} && echo "***function libcheck"
-  libs=( $(awk '{ print $6 }' "/proc/${1}/maps" | grep '/' | sort -u | xargs file | grep ELF | awk '{ print ${1} }' | sed 's/:/ /') )
+  libs=( $(awk '{ print $6 }' "/proc/${1}/maps" | grep '/' | sort -u | xargs file | grep ELF | awk '{ print $1 }' | sed 's/:/ /') )
 
   echo_message "\n* Loaded libraries (file information, # of mapped files: ${#libs[@]}):\n\n" "" "" "\"libs\": {"
 
@@ -1984,4 +1984,4 @@ for variable in CHK_DIR CHK_FILE CHK_FORTIFY_FILE CHK_FORTIFY_PROC CHK_PROC CHK_
 done
 
 # call the function
-${CHK_FUNCTION}
\ No newline at end of file
+${CHK_FUNCTION}
